### PR TITLE
fix: sapphire-localnet: Wait for postgres and the KM to be ready

### LINKED ISSUE #528: docker: Wait for postgres before starting oasis-web3-gateway
Occasionally it happens that oasis-web3-gateway is launched too early and postgres isn't up and running yet. Then `localhost:8545` simply yields `connection refused` for end users.

```
cat /var/log/oasis-web3-gateway.log 
{"addr":"unix:/serverdir/node/net-runner/network/client-0/internal.sock","caller":"main.go:196","level":"info","module":"main","msg":"connecting to local node","ts":"2024-02-22T16:20:06.344504604Z"}
failed to migrate DB: init migrations: dial tcp 127.0.0.1:5432: connect: connection refused
```

Wait for postgres to start up before running oasis-web3-gateay.

### PR DESCRIPTION
**Pull request by abukosek**
Postgres sometimes needs a bit more time to start up, which makes the gateway fail.  We now wait for it before starting the gateway.

Also, the key manager sometimes needs longer to generate the public ephemeral key needed for confidential contracts to work.  We now wait for the epehemeral key to become available before continuing with the rest of the container initialization.

Fixes #528.

### CODE REVIEW

File: docker/common/start.sh [hunk 1]
```diff
@@ -74,12 +74,18 @@ notice "Starting oasis-net-runner with ${CYAN}${PARATIME_NAME}${OFF}...\n"
 /spinup-oasis-stack.sh --log.level ${OASIS_NODE_LOG_LEVEL} 2>1 &>/var/log/spinup-oasis-stack.log &
 OASIS_NODE_PID=$!
 
-notice "Starting postgres...\n"
+notice "Starting postgres"
```
Comment (ptrus) [inline docker/common/start.sh]: Maybe reword it to `"Waiting for Postgres to start..."` to make it consistent with other notices.

**Review by ptrus — APPROVED** [review]
Thanks, this looks really useful!

File: docker/common/start.sh (hunk 1 above)
Comment (abukosek) [reply docker/common/start.sh]: Thanks, I've done that now :)
The dots are added by the waiting, so the output looks like this most of the time:
```
 * Starting oasis-net-runner with sapphire...
 * Waiting for Postgres to start....
 * Waiting for Oasis node to start...
 * Starting oasis-web3-gateway...
 * Bootstrapping network (this might take a minute)...
 * Waiting for key manager...
 * Populating accounts...
```

Comment (ptrus) [reply docker/common/start.sh]: Makes sense :+1: 

Should the hardcoded three dots (`...`) be removed in other "Waiting for..." lines then as well?

Comment (abukosek) [reply docker/common/start.sh]: No, because in some cases it doesn't have to wait at all and then the output looks weird without the dots :D

**Comment by matevz** [comment]
Thanks, I can confirm this fixed the issue with postgres and the KM as well!
